### ISSUE TITLE
Face Recognition Inaccurate

### ISSUE THREAD

**Issue by dlashua** [roflcoopter/viseron#88]
I know viseron only leverages the functionality provided by dlib for Face Recognition. However, it is regularly incorrectly recognizing faces (i.e. there is a face in the stream, but it is not the face that has been detected).

Is there a way to get a confidence score into the binary_sensor for face detection? Or, alternately, a way to only indicate a particular face if the confidence is over a certain value? This would allow me to know whether I should trust the data received. 

Finally, if dlib could produce more debugging information, I'd be able to determine which face image was being matched incorrectly and potentially, remove that image from the reference.

**Comment by roflcoopter** [comment]
I can look into getting more confidence values in the binary_sensor.
It would also be possible to make the sensitivty configurable.

Im thinking of implementing a more sophisticated version of this with perhaps OpenFace.

Its not that simple unfortunately, what is happening is it takes all the images and creates a single model per face, so you cant exactly pin point which image it is matching with.
One thing that could improve accuracy aswell is alignment, which basically means that a face viewed at a slight angle will be morphed into being viewed from the front

**Comment by dlashua** [comment]
Confidence values would be a good start anyway. I don't know what it is about this ONE face that I have registered (with 7 different images for that face) but, almost everyone is detected as THAT face. It's hilarious, but, not very useful. :) I've tried removing certain images with no change in results. Next, I'm just going to delete that face entirely to see if that improves anything.

**Comment by roflcoopter** [comment]
yeah that is definitely not optimal... I find it hard to find good examples on the web on how to perform highly accurate face recognition sadly

**Comment by roflcoopter** [comment]
Closing, for better accuracy you can use DeepStack or CompreFace
